### DISCUSSION TITLE
Introduce intentional delay to video stream

### DISCUSSION THREAD

**Discussion by nickjvturner** [Sunoo/homebridge-camera-ffmpeg#954] [Q&A]
Hey there, is it possible to introduce an intentional delay of x many seconds to a stream using homebridge-camera-ffmpeg?

I would like to have a 'live stream' from my raspberry pi running motion-project (this bit already works)
I would like to have a second stream delayed by 20 seconds.

Any pointers would be greatly appreciated.

**Comment by Sunoo** [comment]
I haven’t tried, but maybe this would work? https://stackoverflow.com/a/63238428

**Reply by nickjvturner** [reply]
Into which field should I try adding these parameters?

Video Source or Encoding Options?

**Reply by Sunoo** [reply]
Oh right, it uses filter_complex and not vfilter, I looked quick. You could try including it in the video source.

**Reply by nickjvturner** [reply]
adding:
-filter_complex "[0]tpad=start_duration=2:start_mode=clone[bg];[bg][1]overlay"

to the Video Source, produces the following error:

[12/23/2020, 4:03:55 PM] [Camera FFmpeg] [Camera-2] FFmpeg exited with code: 1 and signal: null (Error)

Could this be achieved with vfilter only?

**Reply by Sunoo** [reply]
I really don’t have arhat much understanding of this functionality of FFmpeg. If I have some free time at some point, I may take a deeper look.

**Reply by nickjvturner** [reply]
Thank you for taking the time to glance at this, if we can achieve this, I think it would be very useful.

**Reply by nickjvturner** [reply]
I have a raspberry pi running motion project, I have managed to 'subscribe' / receive the stream from this pi zero + camera module using homebridge-camera-ffmpeg it is working beautifully.

Next-up: I want to create a duplicate stream with an intentional 15 second delay!

Here is a copy of my config:
 {
            "name": "Camera FFmpeg",
            "porthttp": 8080,
            "cameras": [
                {
                    "name": "Live Stream",
                    "videoConfig": {
                        "source": "-re -i http://172.16.0.45:9081",
                        "maxStreams": 2,
                        "maxWidth": 1280,
                        "maxHeight": 720,
                        "maxFPS": 30,
                        "debug": false
                    }
                },
                {
                    "name": "Delayed Stream",
                    "videoConfig": {
                        "source": "-re -i http://172.16.0.45:9081",
                        "maxStreams": 2,
                        "maxWidth": 1280,
                        "maxHeight": 720,
                        "maxFPS": 30,
                        "videoFilter": "drawbox=x=10:y=0:w=200:h=60:color=red@0.5,tpad=start_duration=15",
                        "debug": true
                    }
                }

The drawbox is not required, but is there to prove that my video filters are doing something!
tpad seems to blank the delayed stream for 15 seconds, but does not appear to delay the footage once the 15 seconds have passed.

Any suggestions on what to try next will be greatly appreciated.
